test/dotcom/system_status/groups_test.exs

```diff
@@ -0,0 +1,632 @@
+  # for the given route_id with an empty branch_ids list.
+  defp statuses_for(groups, route_id, branch_ids \\ []) do
+    groups
+    |> Enum.find(&(&1.route_id == route_id))
+    |> Map.fetch!(:branches_with_statuses)
+    |> Enum.find(&(&1.branch_ids == branch_ids))
+    |> Map.get(:statuses)
+  end
+
+  # Returns the beginning of the day in the Eastern time zone.
+  defp beginning_of_day() do
+    Timex.beginning_of_day(Timex.now("America/New_York"))
+  end
+
+  # Returns the end of the day in the Eastern time zone.
+  defp end_of_day() do
+    Timex.end_of_day(Timex.now("America/New_York"))
+  end
+
+  # Returns a random time during the day today.
+  defp time_today() do
+    between(beginning_of_day(), end_of_day())
+  end
+
+  # Returns a random time during the day today before the time provided.
+  defp time_before(time) do
+    between(beginning_of_day(), time)
+  end
+
+  # Returns a random time during the day today after the time provided.
+  defp time_after(time) do
+    between(time, end_of_day())
+  end
+
+  # Returns a random time between the times provided in the Eastern time zone.
+  defp between(time1, time2) do
+    Faker.DateTime.between(time1, time2) |> Timex.to_datetime("America/New_York")
+  end
+
+  # Returns a random alert that will be active at the time given by
+  # the required `:time` opt.
+  #
+  # Required opts:
+  #  - route_id
+  #  - time
+  #
+  # Optional opts:
+  #  - effect (default behavior is to choose an effect at random)
+  defp current_alert(opts) do
+    {time, opts} = opts |> Keyword.pop!(:time)
+
+    start_time = time_before(time)
+    end_time = time_after(time)
+
+    opts
+    |> Keyword.put_new(:active_period, [{start_time, end_time}])
+    |> alert()
+  end
+
+  # Returns a random alert whose active_period starts at the provided
+  # `:start_time` opt.
+  #
+  # Required opts:
+  #  - route_id
+  #  - start_time
+  #
+  # Optional opts:
+  #  - effect (default behavior is to choose an effect at random)
+  defp future_alert(opts) do
+    {start_time, opts} = opts |> Keyword.pop!(:start_time)
+
+    opts
+    |> Keyword.put_new(:active_period, [{start_time, time_after(start_time)}])
+    |> alert()
+  end
+
+  # Returns a random alert for the given `:route_id` and
+  # `:active_period` opts.
+  #
+  # Required opts:
+  #  - route_id
+  #  - active_period (Note that this is an array)
+  #
+  # Optional opts:
+  #  - effect (default behavior is to choose an effect at random)
+  defp alert(opts) do
+    route_id = opts |> Keyword.fetch!(:route_id)
+    effect = opts[:effect] || Faker.Util.pick(@effects)
+    active_period = opts |> Keyword.fetch!(:active_period)
+
+    Alert.build(:alert,
+      effect: effect,
+      informed_entity:
+        InformedEntitySet.build(:informed_entity_set,
+          route: route_id,
+          entities: [
+            InformedEntity.build(:informed_entity, route: route_id)
+          ]
+        ),
+      active_period: active_period
+    )
+  end
```

Comment:
Any code that goes toward building an alert should go in the alerts factory.

Reply:
I sort of see that, although exactly what belongs in a factory versus just here eludes me at the moment.

The problem is that I don't want to have to copy-pasta the `informed_entity` / `informed_entity_set` nesting over and over again, but I'm not sure that it makes sense to pull this into a factory as-is either.

There probably is a way to do it gracefully, but unless we can get that taken care of _real fast_, I think I'll ask that we tackle it as a follow-up, since this work is blocking the rest of the project.

Reply:
You can see in this factory: https://github.com/mbta/dotcom/pull/2336/files#diff-8fbc124c966696c04346223e24b3cc686bb63d07f0f3a35316e93f8dd30c6384 how I separated `_factory` functions from other functions.